packages/transformers/js/core/src/lib.rs

```diff
@@ -396,6 +400,34 @@ pub fn transform(config: Config) -> Result<TransformResult, std::io::Error> {
               module.fold_with(&mut passes)
             };
 
+            let mut has_node_replacements = false;
+
+            let module = {
+              let mut passes = chain!(
+                // Insert placeholders for context variables like __dirname and __filename
+                Optional::new(
+                  NodeReplacer {
+                    source_map: &source_map,
+                    items: &mut global_deps,
+                    globals: HashMap::new(),
+                    project_root: Path::new(&config.project_root),
+                    filename: Path::new(&config.filename),
+                    decls: &mut decls,
+                    global_mark,
+                    scope_hoist: config.scope_hoist,
+                    has_node_replacements: &mut has_node_replacements,
+                  },
+                  config.node_replacer
+                ),
+                // Inject SWC helpers if needed.
+                helpers::inject_helpers(),
```

Comment:
1. We don't need to run `inject_helpers` another time
2. I'd just move the NodeReplacer directly between `GlobalReplacer` and `preset_env` directly above this

Reply:
![Bildschirmfoto 2022-04-07 um 11 51 49](https://user-images.githubusercontent.com/16143594/162172649-feb77921-b362-49c0-ab0b-6c8a3d9509ef.png)

![Bildschirmfoto 2022-04-07 um 11 52 00](https://user-images.githubusercontent.com/16143594/162172679-498426d6-eb58-459e-8338-7884a265a15d.png)

I tried that at the beginning but I couldn't figure out how to solve this borrow issue :/

Reply:
There's no way around this borrow issue. You have to do this separately, so just: 
```rust
            let mut has_node_replacements = false;
            let module = module.fold_with(&mut Optional::new(
              NodeReplacer {
                source_map: &source_map,
                items: &mut global_deps,
                globals: HashMap::new(),
                project_root: Path::new(&config.project_root),
                filename: Path::new(&config.filename),
                decls: &mut decls,
                global_mark,
                scope_hoist: config.scope_hoist,
                has_node_replacements: &mut has_node_replacements,
              },
              config.node_replacer,
            ));
```

Reply:
Done 👍